pulp_ansible/app/models.py

```diff
@@ -34,30 +40,76 @@ class Meta:
         )
 
 
-class Collection(Content):
+class Collection(Model):
+    """A model representing a Collection."""
+
+    TYPE = 'collection'
+
+    namespace = models.CharField(max_length=64, editable=False)
+    name = models.CharField(max_length=64, editable=False)
+
+    deprecated = models.BooleanField(default=False)
+
+    class Meta:
+        unique_together = (
+            'namespace',
+            'name',
+        )
+
+
+class CollectionVersion(Content):
     """
     A content type representing a Collection.
+
+    Fields:
+
+        version (models.CharField): Version string.
+        metadata: Artifact metadata in JSON format.
+        contents: Contnet items (e.g. roles, modules) provided by
+            collection in JSON format.
+        quality_score: Total version quality score,
+            floating-point number in range [0, 5].
+
+    Relations:
+
+        collection: Reference to a collection model.
     """
 
-    TYPE = 'collection'
+    TYPE = 'collection_version'
 
-    namespace = models.CharField(max_length=64)
-    name = models.CharField(max_length=64)
-    version = models.CharField(max_length=128)
+    collection = models.ForeignKey(
+        Collection,
+        on_delete=models.CASCADE,
+        related_name='versions',
+        editable=False,
+    )
+    version = models.CharField(max_length=128, editable=False)
+
+    quality_score = models.FloatField(null=True, editable=False)
+    metadata = psql_fields.JSONField(default=dict, editable=False)
```

Comment:
What about making this a `CharField` and before writing to it serialize the json data to a string at the db level? The serializer could continue to read/write json natively.

Reply:
The problem is querying the field. You can't query a TextField the same way. See:

https://docs.djangoproject.com/en/2.2/ref/contrib/postgres/fields/#querying-jsonfield

Reply:
I see what you are saying. @daviddavis so that filtering also works with the filtering at the API level and its `__` syntax too?

@cutwater Is the data stored in this field have the same schema or are there many schemas? Also what types of queries will be sent against this?